client/src/app/core/repositories/users/user-repository.service.ts

```diff
@@ -500,6 +500,9 @@ export class UserRepositoryService extends BaseRepository<ViewUser, User, UserTi
     }
 
     private preventAlterationOnDemoUsers(users: ViewUser | ViewUser[]): void {
+        if (!this.demoModeUserIds) {
+            return;
+        }
```

Comment:
That's dirty. Rather have:
```
if (this.demoModeUserIds) {
    ...
}
```
and do not return anything in void functions

Reply:
-> Early return pattern?

> and do not return anything in void functions

I do not return anything, right?

Reply:
>  Early return pattern?

It's really ugly. You may validate that data is correct, not wrong. I'd prefer the optional chaining method as in the other pr.

> I do not return anything, right?

Do not use the `return` statement in void function